paddle/fluid/framework/parallel_executor.cc

```diff
@@ -150,8 +150,16 @@ void ParallelExecutor::BCastParamsToGPUs(
           buffer = t->mutable_data(place, main_tensor.type());
         }
         auto &nccl_ctx = member_->nccl_ctxs_->at(place);
-        platform::dynload::ncclBcast(buffer, numel, data_type, 0,
-                                     nccl_ctx.comm_, nccl_ctx.stream());
+
+        if (builder_.get() != nullptr &&
```

Comment:
Not sure this change is needed. It seems ```BCastParamsToGPUs``` is only called once at beginning to bcast parameter to each devices. It's not used during training?

Reply:
It's used during training at the end of eatch mini-batch, such as the following code snippet:

https://github.com/PaddlePaddle/Paddle/blob/831909ce6923c9577f5bf75d8fedc6d83f23265f/benchmark/fluid/fluid_benchmark.py#L363-L365